{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"content\"><h3>Content</h3></a>\n",
    "\n",
    "- [Dependencies, paths and datasets upload](#upload)\n",
    "- [Raw dataset consolidation](#raw):\n",
    "    - [MGI-MP associations consolidation from HMD and PhenoGeno](#mouse)\n",
    "    - [Human phenotypes dataset consolidation](#human)\n",
    "    - [Human and mouse datasets combining](#humandmouse)\n",
    "- [Up-leveling HPO and MPO terms using HPO-MPO system level term mapping by MGI](#uplevel)\n",
    "- [Filtering the dataset to monosemantic](#mono)\n",
    "- [Low-level MPO and HPO terms filtering](#lowfilter)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"upload\"><h4>Dependencies, paths and datasets upload</h4></a>\n",
    "[Back to content](#content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "\n",
    "from collections import defaultdict\n",
    "from functools import reduce\n",
    "from operator import or_\n",
    "import itertools\n",
    "import os\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "from low2system import replace_terms_by_level\n",
    "from ontobio.ontol_factory import OntologyFactory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# set paths to databases and ontologies\n",
    "\n",
    "HMD = \"../databases/HMD.rpt\"\n",
    "PHENO_GENO = \"../databases/PhenoGeno.rpt\"\n",
    "HPO = \"../databases/HPO.txt\"\n",
    "HP2MP = \"../databases/HP2MP.tsv\"\n",
    "\n",
    "PATH_HPO = os.path.abspath(\"../ontologies/hpo.json\")\n",
    "PATH_MPO = os.path.abspath(\"../ontologies/mpo.json\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"raw\"><h4>Raw dataset consolidation</h4></a>\n",
    "\n",
    "Consolidating raw database with low-level HPO terms from HPO.txt, low-level MPO terms from PhenoGeno and high-level MPO terms from HMD.\n",
    "\n",
    "<a id=\"mouse\"><h4>MGI-MP associations consolidation from HMD and PhenoGeno</h4></a>\n",
    "\n",
    "[Back to content](#content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MGI with associated phenotypes\n",
      "12577\n",
      "MGI without associated phenotypes\n",
      "7554\n"
     ]
    }
   ],
   "source": [
    "# dict; key is MGI, value are low-level phenotypes from HMD\n",
    "\n",
    "f = open(HMD, \"r\")\n",
    "\n",
    "mouse_mgi_to_HMD_phenotypes = {}\n",
    "\n",
    "for num, line in enumerate(f, 1):\n",
    "    line_data = line.rstrip().split(\"\\t\")\n",
    "    num_fields = len(line_data)\n",
    "    if num_fields == 4:\n",
    "        # no HL MP tags\n",
    "        hl_pheno = None\n",
    "    else:\n",
    "        hl_pheno = set(line_data[-1].split(\", \"))\n",
    "    hum_symbol = line_data[0]\n",
    "    hum_etrez = line_data[1]\n",
    "    mouse_symbol = line_data[2]\n",
    "    mouse_mgi = line_data[3]\n",
    "\n",
    "    mouse_mgi_to_HMD_phenotypes[mouse_mgi] = hl_pheno\n",
    "\n",
    "f.close()\n",
    "\n",
    "print(\"MGI with associated phenotypes\")\n",
    "print(len([k for k, v in mouse_mgi_to_HMD_phenotypes.items() if v is not None]))\n",
    "\n",
    "print(\"MGI without associated phenotypes\")\n",
    "print(len([k for k, v in mouse_mgi_to_HMD_phenotypes.items() if v is None]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# dict; key is MGI, value are system-level phenotypes from PhenoGeno\n",
    "\n",
    "mgi_to_phenogeno = defaultdict(set)\n",
    "\n",
    "f = open(PHENO_GENO, \"r\")\n",
    "for num, line in enumerate(f, 1):\n",
    "    line_data = line.rstrip().split(\"\\t\")\n",
    "    mp = line_data[3]\n",
    "    mgis = line_data[5].split(\"|\")\n",
    "    for mgi in mgis:\n",
    "        mgi_to_phenogeno[mgi].add(mp)\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20131\n",
      "22791\n",
      "12577\n"
     ]
    }
   ],
   "source": [
    "hmd_keys = set(mouse_mgi_to_HMD_phenotypes.keys())\n",
    "phenogeno_keys = set(mgi_to_phenogeno.keys())\n",
    "\n",
    "hmd_keys_not_none = {k: v for k, v in mouse_mgi_to_HMD_phenotypes.items() if v is not None}\n",
    "print(len(hmd_keys))\n",
    "print(len(phenogeno_keys))\n",
    "print(len(hmd_keys_not_none.keys()))\n",
    "\n",
    "hmd_keys_not_none_set = set(hmd_keys_not_none.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22808\n"
     ]
    }
   ],
   "source": [
    "mgi_genes_union = reduce(or_, [hmd_keys_not_none_set, phenogeno_keys])\n",
    "print(len(mgi_genes_union))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# intermediate dataset creation\n",
    "# dataset contains MGI, MPO terms from HMD, MPO terms from PhenoGeno\n",
    "\n",
    "f = open(\"../intermediate_data/mgi_mp_hmd_pg.tsv\", \"w\")\n",
    "f.write(\"MGI\\tHMD\\tPhenoGeno\\n\")\n",
    "for mgi in mgi_genes_union:\n",
    "    hmd_terms = mouse_mgi_to_HMD_phenotypes.get(mgi, None)\n",
    "    phenogeno_terms = mgi_to_phenogeno.get(mgi, None)\n",
    "\n",
    "    hmd_terms = hmd_terms if hmd_terms else {\"NA\"}\n",
    "    phenogeno_terms = phenogeno_terms if phenogeno_terms else {\"NA\"}\n",
    "\n",
    "    hmd_terms = \",\".join(hmd_terms)\n",
    "    phenogeno_terms = \",\".join(phenogeno_terms)\n",
    "\n",
    "    f.write(f\"{mgi}\\t{hmd_terms}\\t{phenogeno_terms}\\n\")\n",
    "f.close()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"human\"><h4>Human phenotypes dataset consolidation</h4></a>\n",
    "\n",
    "[Back to content](#content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read main table with human and mouse genes into a pandas dataframe\n",
    "\n",
    "human_mouse_genes = pd.read_csv(HMD, delimiter=\"\\t\", header=None)\\\n",
    "    .rename(columns={0: \"gene_human\", 1: \"entrez_id_human\", 2: \"gene_mouse\", 3: \"MGI\", 4: \"MP\"}).drop(5, axis=1)\n",
    "\n",
    "human_HPO_table = pd.read_csv(HPO, delimiter=\"\\t\", skiprows=1, header=None, usecols=range(0,4))\\\n",
    "   .rename(columns={0: \"entrez_id_human\", 1: \"entrez-gene-symboln\", 2: \"HPO-Term-ID\", 3: \"HPO-Term-Name\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# combine genes by HPO terms and HPO terms names\n",
    "\n",
    "human_HPO_table = human_HPO_table\\\n",
    "    .groupby(\"entrez_id_human\")\\\n",
    "    .agg({\"HPO-Term-ID\": list, \"HPO-Term-Name\": list})\\\n",
    "    .reset_index()\n",
    "\n",
    "# concatenate the list of HPOs into a string variable\n",
    "\n",
    "human_HPO_table[\"HPO-Term-ID\"] = human_HPO_table[\"HPO-Term-ID\"].apply(lambda x: ', '.join(x))\n",
    "human_HPO_table[\"HPO-Term-Name\"] = human_HPO_table[\"HPO-Term-Name\"].apply(lambda x: ', '.join(x))\n",
    "\n",
    "# combine the main table with the genes of both mice and humans \n",
    "# and the table with the human phenotype by Entrez Gene ID\n",
    "\n",
    "hpo_db = pd.merge(human_mouse_genes, human_HPO_table, how='left', on = 'entrez_id_human')\\\n",
    "    .rename(columns={\"HPO-Term-ID\": \"HP_low_level\"})[[\"entrez_id_human\", \"HP_low_level\", \"MGI\"]]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"humandmouse\"><h4>Human and mouse datasets combining</h4></a>\n",
    "\n",
    "Combine ortholodues info (HMD), mouse phenotype info (mgi_mp_hmd_pg.tsv) and human phenotype info (hpo_db).\n",
    "\n",
    "[Back to content](#content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "mp_db = pd.read_csv(\"../intermediate_data/mgi_mp_hmd_pg.tsv\", delimiter=\"\\t\")\n",
    "hmd_db = pd.read_csv(\"../databases/HMD.rpt\", delimiter=\"\\t\", header=None)\\\n",
    "    .rename(columns={0: \"gene_human\", 1: \"entrez_id_human\", 2: \"gene_mouse\", 3: \"MGI\"})\\\n",
    "    .drop([4, 5], axis=1)\n",
    "\n",
    "db_with_mp = pd.merge(hmd_db, mp_db, on=\"MGI\", how=\"left\")\n",
    "db_with_mp_hp = pd.merge(db_with_mp, hpo_db, on=[\"entrez_id_human\", \"MGI\"], how=\"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_human</th>\n",
       "      <th>entrez_id_human</th>\n",
       "      <th>gene_mouse</th>\n",
       "      <th>MGI</th>\n",
       "      <th>HMD</th>\n",
       "      <th>PhenoGeno</th>\n",
       "      <th>HP_low_level</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>1</td>\n",
       "      <td>A1bg</td>\n",
       "      <td>MGI:2152878</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>29974</td>\n",
       "      <td>A1cf</td>\n",
       "      <td>MGI:1917115</td>\n",
       "      <td>MP:0005378,MP:0005397,MP:0005376,MP:0005367,MP...</td>\n",
       "      <td>MP:0002083,MP:0003222,MP:0003061,MP:0004231,MP...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>2</td>\n",
       "      <td>A2m</td>\n",
       "      <td>MGI:2449119</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>127550</td>\n",
       "      <td>A3galt2</td>\n",
       "      <td>MGI:2685279</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>53947</td>\n",
       "      <td>A4galt</td>\n",
       "      <td>MGI:3512453</td>\n",
       "      <td>MP:0005386,MP:0010768,MP:0005376</td>\n",
       "      <td>MP:0009747,MP:0008874,MP:0009767</td>\n",
       "      <td>HP:0000006, HP:0010970</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_human  entrez_id_human gene_mouse          MGI  \\\n",
       "0       A1BG                1       A1bg  MGI:2152878   \n",
       "1       A1CF            29974       A1cf  MGI:1917115   \n",
       "2        A2M                2        A2m  MGI:2449119   \n",
       "3    A3GALT2           127550    A3galt2  MGI:2685279   \n",
       "4     A4GALT            53947     A4galt  MGI:3512453   \n",
       "\n",
       "                                                 HMD  \\\n",
       "0                                                 []   \n",
       "1  MP:0005378,MP:0005397,MP:0005376,MP:0005367,MP...   \n",
       "2                                                 []   \n",
       "3                                                 []   \n",
       "4                   MP:0005386,MP:0010768,MP:0005376   \n",
       "\n",
       "                                           PhenoGeno            HP_low_level  \n",
       "0                                                 []                      []  \n",
       "1  MP:0002083,MP:0003222,MP:0003061,MP:0004231,MP...                      []  \n",
       "2                                                 []                      []  \n",
       "3                                                 []                      []  \n",
       "4                   MP:0009747,MP:0008874,MP:0009767  HP:0000006, HP:0010970  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db_with_mp_hp = db_with_mp_hp.fillna(\"[]\") # for more convenient to file writing\n",
    "db_with_mp_hp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# intermediate semi-final dataset with low-level MPO and HPO terms\n",
    "\n",
    "db_with_mp_hp.to_csv(\"../intermediate_data/semi_final_db.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# semi-final dataset written in a proper way\n",
    "\n",
    "f_in = open(\"../intermediate_data/semi_final_db.tsv\", \"r\")\n",
    "f_in.__next__()\n",
    "f_out = open(\"../intermediate_data/Human_Mouse_genes_phenotypes_raw.tsv\", \"w\")\n",
    "f_out.write(\"gene_human\\tentrez_id_human\\tgene_mouse\\tMGI\\tMP_HMD\\tMP_PhenoGeno\\tHP_genes_to_phenotype\\n\")\n",
    "\n",
    "for line in f_in:\n",
    "    line_data = line.rstrip().split(\"\\t\")\n",
    "    gene_hum = line_data[1]\n",
    "    entrez_id = line_data[2]\n",
    "    gene_mouse = line_data[3]\n",
    "    mgi = line_data[4]\n",
    "    mp_hmd = line_data[5]\n",
    "    mp_hmd_to_wr = \",\".join(mp_hmd.strip(\"[]\").split(\",\"))\n",
    "    mp_pg = line_data[6]\n",
    "    mp_pg_to_wr = \",\".join(mp_pg.strip(\"[]\").split(\",\"))\n",
    "    hp_gtp = line_data[7]\n",
    "    hp_gtp_to_wr = \",\".join(hp_gtp.strip(\"[]\").split(\", \"))\n",
    "    f_out.write(f\"{gene_hum}\\t{entrez_id}\\t{gene_mouse}\\t{mgi}\\t{mp_hmd_to_wr}\\t{mp_pg_to_wr}\\t{hp_gtp_to_wr}\\n\")\n",
    "\n",
    "f_in.close()\n",
    "f_out.close()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"uplevel\"><h4>Up-leveling HPO and MPO terms using HPO-MPO system level term mapping by MGI</h4></a>\n",
    "\n",
    "Adding to the database columns with system-level HPO and MPO terms and rewriting to file.\n",
    "\n",
    "[Back to content](#content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "semi_final = pd.read_csv(\"../intermediate_data/Human_Mouse_genes_phenotypes_raw.tsv\", delimiter=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HPO terms count 94\n",
      "MPO terms count 26\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'HP:0001871': ['MP:0005397'],\n",
       " 'HP:0004377': ['MP:0002006'],\n",
       " 'HP:0009124': ['MP:0005375'],\n",
       " 'HP:0100494': ['MP:0005397', 'MP:0005387'],\n",
       " 'HP:3000050': ['MP:0005382'],\n",
       " 'HP:0100685': ['MP:0005390'],\n",
       " 'HP:0100536': ['MP:0010771'],\n",
       " 'HP:0100658': ['MP:0005387'],\n",
       " 'HP:0100881': ['MP:0002006'],\n",
       " 'HP:0100898': ['MP:0010771'],\n",
       " 'HP:0001371': ['MP:0005390'],\n",
       " 'HP:0100790': ['MP:0005378'],\n",
       " 'HP:0100699': ['MP:0005376'],\n",
       " 'HP:0000234': ['MP:0005382'],\n",
       " 'HP:0000464': ['MP:0005378'],\n",
       " 'HP:0040064': ['MP:0005371'],\n",
       " 'HP:0001939': ['MP:0005376'],\n",
       " 'HP:0000816': ['MP:0005384'],\n",
       " 'HP:0040127': ['MP:0010771', 'MP:0005379'],\n",
       " 'HP:0025354': ['MP:0005384'],\n",
       " 'HP:0003565': ['MP:0005397'],\n",
       " 'HP:0001787': ['MP:0005389'],\n",
       " 'HP:0001194': ['MP:0005380'],\n",
       " 'HP:0001560': ['MP:0005380'],\n",
       " 'HP:0011425': ['MP:0005380'],\n",
       " 'HP:0001789': ['MP:0005376'],\n",
       " 'HP:0010880': ['MP:0005380'],\n",
       " 'HP:0001622': ['MP:0005389'],\n",
       " 'HP:0002686': ['MP:0005389'],\n",
       " 'HP:0001557': ['MP:0005386'],\n",
       " 'HP:0003270': ['MP:0005378'],\n",
       " 'HP:0030142': ['MP:0005381'],\n",
       " 'HP:0002039': ['MP:0005386'],\n",
       " 'HP:0002019': ['MP:0005381'],\n",
       " 'HP:0002014': ['MP:0005381'],\n",
       " 'HP:0011968': ['MP:0005386'],\n",
       " 'HP:0002017': ['MP:0005381'],\n",
       " 'HP:0004396': ['MP:0005386'],\n",
       " 'HP:0002038': ['MP:0005386'],\n",
       " 'HP:0011620': ['MP:0005378'],\n",
       " 'HP:0011024': ['MP:0005381'],\n",
       " 'HP:0001392': ['MP:0005370'],\n",
       " 'HP:0001732': ['MP:0005379'],\n",
       " 'HP:0002585': ['MP:0005378'],\n",
       " 'HP:0001743': ['MP:0005397', 'MP:0005387'],\n",
       " 'HP:0001541': ['MP:0005376'],\n",
       " 'HP:0005217': ['MP:0005378'],\n",
       " 'HP:0003271': ['MP:0005378'],\n",
       " 'HP:0004298': ['MP:0005378'],\n",
       " 'HP:0000775': ['MP:0005369'],\n",
       " 'HP:0100016': ['MP:0005378'],\n",
       " 'HP:0000769': ['MP:0010771'],\n",
       " 'HP:0100013': ['MP:0002006'],\n",
       " 'HP:0001626': ['MP:0005385'],\n",
       " 'HP:0000598': ['MP:0005377'],\n",
       " 'HP:0000818': ['MP:0005379'],\n",
       " 'HP:0003117': ['MP:0005376'],\n",
       " 'HP:0012029': ['MP:0005376'],\n",
       " 'HP:0100568': ['MP:0002006'],\n",
       " 'HP:0000478': ['MP:0005391'],\n",
       " 'HP:0000078': ['MP:0005389'],\n",
       " 'HP:0000079': ['MP:0005367'],\n",
       " 'HP:0012620': ['MP:0005380', 'MP:0005367', 'MP:0005381'],\n",
       " 'HP:0007379': ['MP:0002006'],\n",
       " 'HP:0100589': ['MP:0005389', 'MP:0005367'],\n",
       " 'HP:0002715': ['MP:0005387'],\n",
       " 'HP:0001574': ['MP:0010771'],\n",
       " 'HP:0003011': ['MP:0005369'],\n",
       " 'HP:0008316': ['MP:0005384'],\n",
       " 'HP:0009728': ['MP:0002006'],\n",
       " 'HP:0008985': ['MP:0005375'],\n",
       " 'HP:0000707': ['MP:0003631'],\n",
       " 'HP:0002086': ['MP:0005388'],\n",
       " 'HP:0100606': ['MP:0002006'],\n",
       " 'HP:0000924': ['MP:0005390'],\n",
       " 'HP:0010766': ['MP:0005376'],\n",
       " 'HP:0045027': ['MP:0005378'],\n",
       " 'HP:0001608': ['MP:0005386'],\n",
       " 'HP:0001507': ['MP:0005378'],\n",
       " 'HP:0002664': ['MP:0002006'],\n",
       " 'HP:0000708': ['MP:0005386'],\n",
       " 'HP:0025406': ['MP:0005386'],\n",
       " 'HP:0031352': ['MP:0005378'],\n",
       " 'HP:0025143': ['MP:0005376'],\n",
       " 'HP:0012378': ['MP:0005386'],\n",
       " 'HP:0100812': ['MP:0005382'],\n",
       " 'HP:0031217': ['MP:0005376'],\n",
       " 'HP:0031058': ['MP:0005386'],\n",
       " 'HP:0002607': ['MP:0005381'],\n",
       " 'HP:0000020': ['MP:0005367'],\n",
       " 'HP:0030166': ['MP:0005379'],\n",
       " 'HP:0012531': ['MP:0005386'],\n",
       " 'HP:0025144': ['MP:0005386'],\n",
       " 'HP:0040006': ['MP:0010768'],\n",
       " '': ['']}"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# uploading dataset on HPO to MPO mapping\n",
    "# dictionary creation (key = HPO term, value = list pf MPO terms)\n",
    "# empty key-empty value in dict is for convenience for empty fields\n",
    "\n",
    "mapping_by_mp = pd.read_csv(HP2MP, delimiter=\"\\t\")\n",
    "print(f\"HPO terms count {len(set(mapping_by_mp.subject_id.tolist()))}\")\n",
    "print(f\"MPO terms count {len(set(mapping_by_mp.object_id.tolist()))}\")\n",
    "\n",
    "mapping_system_level = dict()\n",
    "for pair in zip(mapping_by_mp.subject_id, mapping_by_mp.object_id):\n",
    "    if pair[0].strip() not in mapping_system_level.keys():\n",
    "        mapping_system_level[pair[0].strip()] = [pair[1].strip()]\n",
    "    else:\n",
    "        mapping_system_level[pair[0].strip()].append(pair[1].strip())\n",
    "mapping_system_level[\"\"] = [\"\"]\n",
    "mapping_system_level"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# creating lists of system-level HPO and MPO terms\n",
    "# for low2system transpose\n",
    "\n",
    "hpo_system_level = list(mapping_system_level.keys())\n",
    "list_mpo = mapping_system_level.values()\n",
    "mpo_system_level = list(set(list(itertools.chain.from_iterable(list_mpo))))\n",
    "\n",
    "# for more convenient low2system transpose\n",
    "\n",
    "semi_final = semi_final.fillna('') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# adding new columns to semi-final dataset with system HPO and MPO terms\n",
    "\n",
    "semi_final[\"MP_system_level\"] = semi_final.MP_PhenoGeno\\\n",
    "    .apply(lambda x: replace_terms_by_level(x.split(\",\"), \"mp\", mpo_system_level))\n",
    "\n",
    "semi_final[\"HP_system_level\"] = semi_final.HP_genes_to_phenotype\\\n",
    "    .apply(lambda x: replace_terms_by_level(x.split(\",\"), \"hp\", hpo_system_level))\n",
    "\n",
    "# adding column with system HPO mapped on system MPO terms\n",
    "\n",
    "semi_final[\"MP_from_HP_system_level\"] = semi_final.HP_system_level\\\n",
    "    .apply(lambda x: [mapping_system_level[i] for i in x])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_human</th>\n",
       "      <th>entrez_id_human</th>\n",
       "      <th>gene_mouse</th>\n",
       "      <th>MGI</th>\n",
       "      <th>MP_HMD</th>\n",
       "      <th>MP_PhenoGeno</th>\n",
       "      <th>HP_genes_to_phenotype</th>\n",
       "      <th>MP_system_level</th>\n",
       "      <th>HP_system_level</th>\n",
       "      <th>MP_from_HP_system_level</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>1</td>\n",
       "      <td>A1bg</td>\n",
       "      <td>MGI:2152878</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[[]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>29974</td>\n",
       "      <td>A1cf</td>\n",
       "      <td>MGI:1917115</td>\n",
       "      <td>MP:0005378,MP:0005397,MP:0005376,MP:0005367,MP...</td>\n",
       "      <td>MP:0002083,MP:0003222,MP:0003061,MP:0004231,MP...</td>\n",
       "      <td></td>\n",
       "      <td>[MP:0005378, MP:0003631, MP:0005397, MP:000537...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[[]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>2</td>\n",
       "      <td>A2m</td>\n",
       "      <td>MGI:2449119</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[[]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>127550</td>\n",
       "      <td>A3galt2</td>\n",
       "      <td>MGI:2685279</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[[]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>53947</td>\n",
       "      <td>A4galt</td>\n",
       "      <td>MGI:3512453</td>\n",
       "      <td>MP:0005386,MP:0010768,MP:0005376</td>\n",
       "      <td>MP:0009747,MP:0008874,MP:0009767</td>\n",
       "      <td>HP:0000006,HP:0010970</td>\n",
       "      <td>[MP:0005386, MP:0010768, MP:0005376]</td>\n",
       "      <td>[HP:0001871]</td>\n",
       "      <td>[[MP:0005397]]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_human  entrez_id_human gene_mouse          MGI  \\\n",
       "0       A1BG                1       A1bg  MGI:2152878   \n",
       "1       A1CF            29974       A1cf  MGI:1917115   \n",
       "2        A2M                2        A2m  MGI:2449119   \n",
       "3    A3GALT2           127550    A3galt2  MGI:2685279   \n",
       "4     A4GALT            53947     A4galt  MGI:3512453   \n",
       "\n",
       "                                              MP_HMD  \\\n",
       "0                                                      \n",
       "1  MP:0005378,MP:0005397,MP:0005376,MP:0005367,MP...   \n",
       "2                                                      \n",
       "3                                                      \n",
       "4                   MP:0005386,MP:0010768,MP:0005376   \n",
       "\n",
       "                                        MP_PhenoGeno  HP_genes_to_phenotype  \\\n",
       "0                                                                             \n",
       "1  MP:0002083,MP:0003222,MP:0003061,MP:0004231,MP...                          \n",
       "2                                                                             \n",
       "3                                                                             \n",
       "4                   MP:0009747,MP:0008874,MP:0009767  HP:0000006,HP:0010970   \n",
       "\n",
       "                                     MP_system_level HP_system_level  \\\n",
       "0                                                 []              []   \n",
       "1  [MP:0005378, MP:0003631, MP:0005397, MP:000537...              []   \n",
       "2                                                 []              []   \n",
       "3                                                 []              []   \n",
       "4               [MP:0005386, MP:0010768, MP:0005376]    [HP:0001871]   \n",
       "\n",
       "  MP_from_HP_system_level  \n",
       "0                    [[]]  \n",
       "1                    [[]]  \n",
       "2                    [[]]  \n",
       "3                    [[]]  \n",
       "4          [[MP:0005397]]  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "semi_final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "semi_final.to_csv(\"../intermediate_data/semi_final_system.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# final dataset writing in a covenient format without brackets and quotes\n",
    "\n",
    "f_in = open(\"../intermediate_data/semi_final_system.tsv\", \"r\")\n",
    "f_in.__next__()\n",
    "f_out = open(\"../databases/human_mouse_GPO.tsv\", \"w\")\n",
    "f_out.write(\"gene_human\\tentrez_id_human\\tgene_mouse\\tMGI\\tMP_HMD\\tMP_PhenoGeno\\tHP_genes_to_phenotype\\tMP_system_level\\tHP_system_level\\tMP_from_HP_system_level\\n\")\n",
    "\n",
    "for line in f_in:\n",
    "    line_data = line.rstrip().split(\"\\t\")\n",
    "    gene_hum = line_data[1]\n",
    "    entrez_id = line_data[2]\n",
    "    gene_mouse = line_data[3]\n",
    "    mgi = line_data[4]\n",
    "    mp_hmd = line_data[5]\n",
    "    mp_pg = line_data[6]\n",
    "    hp_gtp = line_data[7]\n",
    "    mp_syst_lvl = \",\".join([i.strip(\"'\") for i in line_data[8].strip(\"[]\").split(\", \")])\n",
    "    hp_syst_lvl = \",\".join([i.strip(\"'\") for i in line_data[9].strip(\"[]\").split(\", \")])\n",
    "    mp_from_hp_sysl_lvl = \",\".join(set([i.strip(\"[]'\") for i in line_data[10][1:-1].split(\", \")]))\n",
    "    f_out.write(f\"{gene_hum}\\t{entrez_id}\\t{gene_mouse}\\t{mgi}\\t{mp_hmd}\\t{mp_pg}\\t{hp_gtp}\\t{mp_syst_lvl}\\t{hp_syst_lvl}\\t{mp_from_hp_sysl_lvl}\\n\")\n",
    "\n",
    "f_in.close()\n",
    "f_out.close()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"mono\"><h4>Filtering the dataset to monosemantic</h4></a>\n",
    "\n",
    "Gene filtering so that only monosemantic genes (having only one orthologue in each organism) are present.\n",
    "\n",
    "[Back to content](#content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_human</th>\n",
       "      <th>entrez_id_human</th>\n",
       "      <th>gene_mouse</th>\n",
       "      <th>MGI</th>\n",
       "      <th>MP_HMD</th>\n",
       "      <th>MP_PhenoGeno</th>\n",
       "      <th>HP_genes_to_phenotype</th>\n",
       "      <th>MP_system_level</th>\n",
       "      <th>HP_system_level</th>\n",
       "      <th>MP_from_HP_system_level</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>1</td>\n",
       "      <td>A1bg</td>\n",
       "      <td>MGI:2152878</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>29974</td>\n",
       "      <td>A1cf</td>\n",
       "      <td>MGI:1917115</td>\n",
       "      <td>MP:0005378,MP:0005397,MP:0005376,MP:0005367,MP...</td>\n",
       "      <td>MP:0002083,MP:0003222,MP:0003061,MP:0004231,MP...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MP:0005378,MP:0003631,MP:0005397,MP:0005376,MP...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>2</td>\n",
       "      <td>A2m</td>\n",
       "      <td>MGI:2449119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>127550</td>\n",
       "      <td>A3galt2</td>\n",
       "      <td>MGI:2685279</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>53947</td>\n",
       "      <td>A4galt</td>\n",
       "      <td>MGI:3512453</td>\n",
       "      <td>MP:0005386,MP:0010768,MP:0005376</td>\n",
       "      <td>MP:0009747,MP:0008874,MP:0009767</td>\n",
       "      <td>HP:0000006,HP:0010970</td>\n",
       "      <td>MP:0005386,MP:0010768,MP:0005376</td>\n",
       "      <td>HP:0001871</td>\n",
       "      <td>MP:0005397</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_human  entrez_id_human gene_mouse          MGI   \n",
       "0       A1BG                1       A1bg  MGI:2152878  \\\n",
       "1       A1CF            29974       A1cf  MGI:1917115   \n",
       "2        A2M                2        A2m  MGI:2449119   \n",
       "3    A3GALT2           127550    A3galt2  MGI:2685279   \n",
       "4     A4GALT            53947     A4galt  MGI:3512453   \n",
       "\n",
       "                                              MP_HMD   \n",
       "0                                                NaN  \\\n",
       "1  MP:0005378,MP:0005397,MP:0005376,MP:0005367,MP...   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                   MP:0005386,MP:0010768,MP:0005376   \n",
       "\n",
       "                                        MP_PhenoGeno  HP_genes_to_phenotype   \n",
       "0                                                NaN                    NaN  \\\n",
       "1  MP:0002083,MP:0003222,MP:0003061,MP:0004231,MP...                    NaN   \n",
       "2                                                NaN                    NaN   \n",
       "3                                                NaN                    NaN   \n",
       "4                   MP:0009747,MP:0008874,MP:0009767  HP:0000006,HP:0010970   \n",
       "\n",
       "                                     MP_system_level HP_system_level   \n",
       "0                                                NaN             NaN  \\\n",
       "1  MP:0005378,MP:0003631,MP:0005397,MP:0005376,MP...             NaN   \n",
       "2                                                NaN             NaN   \n",
       "3                                                NaN             NaN   \n",
       "4                   MP:0005386,MP:0010768,MP:0005376      HP:0001871   \n",
       "\n",
       "  MP_from_HP_system_level  \n",
       "0                     NaN  \n",
       "1                     NaN  \n",
       "2                     NaN  \n",
       "3                     NaN  \n",
       "4              MP:0005397  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gpo = pd.read_csv(\n",
    "    \"../databases/human_mouse_GPO.tsv\", delimiter=\"\\t\"\n",
    ")\n",
    "\n",
    "gpo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "human_mouse_GPO monosematic: 16985\n"
     ]
    }
   ],
   "source": [
    "# Filtering the database by monosemantic feature and\n",
    "# Creating the monosemantic dataset\n",
    "\n",
    "gene_count_human = (\n",
    "    pd.DataFrame(gpo.entrez_id_human.value_counts())\n",
    "    .reset_index()\n",
    "    .rename(columns={\"count\": \"count_human\"})\n",
    ")\n",
    "\n",
    "gene_count_mouse = (\n",
    "    pd.DataFrame(gpo.MGI.value_counts())\n",
    "    .reset_index()\n",
    "    .rename(columns={\"count\": \"count_mouse\"})\n",
    ")\n",
    "\n",
    "gpo_gene_count = gpo.merge(\n",
    "    gene_count_human, on=\"entrez_id_human\", how=\"left\"\n",
    ").merge(gene_count_mouse, on=\"MGI\", how=\"left\")\n",
    "\n",
    "gpo_mono = gpo_gene_count[\n",
    "    (gpo_gene_count.count_human == 1) & (gpo_gene_count.count_mouse == 1)\n",
    "]\n",
    "\n",
    "gpo_mono = gpo_mono.drop([\"count_human\", \"count_mouse\"], axis=1)\n",
    "\n",
    "gpo_mono.to_csv(\"../databases/human_mouse_GPO_mono.tsv\", sep=\"\\t\", index=False)\n",
    "\n",
    "print(f\"human_mouse_GPO monosematic: {gpo_mono.shape[0]}\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"lowfilter\"><h4>Low-level MPO and HPO terms filtering</h4></a>\n",
    "\n",
    "Filtering MPO and HPO low-level terms so that only children of upper-level terms used in HPO-MPO mapping by MGI are present.\n",
    "\n",
    "[Back to content](#content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "hpo = OntologyFactory().create(handle=PATH_HPO)\n",
    "mpo = OntologyFactory().create(handle=PATH_MPO)\n",
    "\n",
    "gpo_mono = pd.read_csv(\"../databases/human_mouse_GPO_mono.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Creating lists of unique MPO and HPO system terms that present in MGI mapping\n",
    "\n",
    "# Lists of unique upper-level MPO and HPO terms from mapping\n",
    "hpo_terms = list(set([term.strip() for term in mapping_by_mp.subject_id.tolist()]))\n",
    "mpo_terms = list(set([term.strip() for term in mapping_by_mp.object_id.tolist()]))\n",
    "\n",
    "# Lists of all descendants for mapped upper-level terms\n",
    "mpo_lower_terms = list()\n",
    "hpo_lower_terms = list()\n",
    "\n",
    "# Add upper-level MPO and HPO terms to the list\n",
    "mpo_lower_terms.extend(mpo_terms)\n",
    "hpo_lower_terms.extend(hpo_terms)\n",
    "\n",
    "for term in mpo_terms:\n",
    "    mpo_lower_terms.extend(mpo.descendants(term))\n",
    "\n",
    "for term in hpo_terms:\n",
    "    hpo_lower_terms.extend(hpo.descendants(term))\n",
    "\n",
    "# Sets with unique lower-level terms that are descendants of the mapped upper-level ones\n",
    "mpo_lower = set(mpo_lower_terms)\n",
    "hpo_lower = set(hpo_lower_terms)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mapped_low(terms_initial, terms_allowed):\n",
    "    \"\"\"\n",
    "    Takes string of initial terms, comma without space separated,\n",
    "    set of allowed terms\n",
    "    returns string of comma without space separated \n",
    "    terms that lie in the intersection of initial and allowed terms \n",
    "    \"\"\"\n",
    "    terms_initial = set(terms_initial.split(\",\"))\n",
    "    terms_allowed = set(terms_allowed)\n",
    "    terms_result = \",\".join(list(terms_initial.intersection(terms_allowed)))\n",
    "    return terms_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def isnan(x):\n",
    "    return x != x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "gpo_mono[\"MPO_lower\"] = gpo_mono.MP_PhenoGeno\\\n",
    "    .apply(lambda x: mapped_low(x, mpo_lower_terms) if not isnan(x) else x)\n",
    "gpo_mono[\"HPO_lower\"] = gpo_mono.HP_genes_to_phenotype\\\n",
    "    .apply(lambda x: mapped_low(x, hpo_lower_terms) if not isnan(x) else x)\n",
    "\n",
    "gpo_mono.to_csv(\"../databases/human_mouse_GPO_v3.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_human</th>\n",
       "      <th>entrez_id_human</th>\n",
       "      <th>gene_mouse</th>\n",
       "      <th>MGI</th>\n",
       "      <th>MP_HMD</th>\n",
       "      <th>MP_PhenoGeno</th>\n",
       "      <th>HP_genes_to_phenotype</th>\n",
       "      <th>MP_system_level</th>\n",
       "      <th>HP_system_level</th>\n",
       "      <th>MP_from_HP_system_level</th>\n",
       "      <th>MPO_lower</th>\n",
       "      <th>HPO_lower</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>1</td>\n",
       "      <td>A1bg</td>\n",
       "      <td>MGI:2152878</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>29974</td>\n",
       "      <td>A1cf</td>\n",
       "      <td>MGI:1917115</td>\n",
       "      <td>MP:0005370,MP:0005376,MP:0005385,MP:0005397,MP...</td>\n",
       "      <td>MP:0006142,MP:0002833,MP:0004090,MP:0004084,MP...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MP:0005370,MP:0005376,MP:0005385,MP:0005389,MP...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MP:0010754,MP:0010052,MP:0008772,MP:0006142,MP...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>2</td>\n",
       "      <td>A2m</td>\n",
       "      <td>MGI:2449119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>127550</td>\n",
       "      <td>A3galt2</td>\n",
       "      <td>MGI:2685279</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>53947</td>\n",
       "      <td>A4galt</td>\n",
       "      <td>MGI:3512453</td>\n",
       "      <td>MP:0010768,MP:0005376,MP:0005386</td>\n",
       "      <td>MP:0009747,MP:0008874,MP:0009767</td>\n",
       "      <td>HP:0000006,HP:0010970</td>\n",
       "      <td>MP:0010768,MP:0005376,MP:0005386</td>\n",
       "      <td>HP:0001871</td>\n",
       "      <td>MP:0005397</td>\n",
       "      <td>MP:0009747,MP:0008874,MP:0009767</td>\n",
       "      <td>HP:0010970</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_human  entrez_id_human gene_mouse          MGI   \n",
       "0       A1BG                1       A1bg  MGI:2152878  \\\n",
       "1       A1CF            29974       A1cf  MGI:1917115   \n",
       "2        A2M                2        A2m  MGI:2449119   \n",
       "3    A3GALT2           127550    A3galt2  MGI:2685279   \n",
       "4     A4GALT            53947     A4galt  MGI:3512453   \n",
       "\n",
       "                                              MP_HMD   \n",
       "0                                                NaN  \\\n",
       "1  MP:0005370,MP:0005376,MP:0005385,MP:0005397,MP...   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                   MP:0010768,MP:0005376,MP:0005386   \n",
       "\n",
       "                                        MP_PhenoGeno  HP_genes_to_phenotype   \n",
       "0                                                NaN                    NaN  \\\n",
       "1  MP:0006142,MP:0002833,MP:0004090,MP:0004084,MP...                    NaN   \n",
       "2                                                NaN                    NaN   \n",
       "3                                                NaN                    NaN   \n",
       "4                   MP:0009747,MP:0008874,MP:0009767  HP:0000006,HP:0010970   \n",
       "\n",
       "                                     MP_system_level HP_system_level   \n",
       "0                                                NaN             NaN  \\\n",
       "1  MP:0005370,MP:0005376,MP:0005385,MP:0005389,MP...             NaN   \n",
       "2                                                NaN             NaN   \n",
       "3                                                NaN             NaN   \n",
       "4                   MP:0010768,MP:0005376,MP:0005386      HP:0001871   \n",
       "\n",
       "  MP_from_HP_system_level                                          MPO_lower   \n",
       "0                     NaN                                                NaN  \\\n",
       "1                     NaN  MP:0010754,MP:0010052,MP:0008772,MP:0006142,MP...   \n",
       "2                     NaN                                                NaN   \n",
       "3                     NaN                                                NaN   \n",
       "4              MP:0005397                   MP:0009747,MP:0008874,MP:0009767   \n",
       "\n",
       "    HPO_lower  \n",
       "0         NaN  \n",
       "1         NaN  \n",
       "2         NaN  \n",
       "3         NaN  \n",
       "4  HP:0010970  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gpo_mono.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10145"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# How many genes have not allowed low-level MPO terms\n",
    "\n",
    "(gpo_mono.MP_PhenoGeno\\\n",
    "    .apply(lambda x: len(x) if not isnan(x) else x) \n",
    " == gpo_mono.MPO_lower\\\n",
    "    .apply(lambda x: len(x) if not isnan(x) else x)).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "213"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# How many genes have not allowed low-level MPO terms\n",
    "\n",
    "(gpo_mono.HP_genes_to_phenotype\\\n",
    "    .apply(lambda x: len(x) if not isnan(x) else x) \n",
    " == gpo_mono.HPO_lower\\\n",
    "    .apply(lambda x: len(x) if not isnan(x) else x)).sum()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "phenotype_to_genotype",
   "language": "python",
   "name": "phenotype_to_genotype"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
